intera_interface/src/intera_interface/limb.py

```diff
@@ -328,6 +375,28 @@ def endpoint_effort(self):
         """
         return deepcopy(self._cartesian_effort)
 
+    def _tip_index(self, tipName):
+        if tipName is None:
+            indx = 0
+        else:
+            try:
+                indx = self._tip_states.names.index(tipName)
+            except ValueError:
+                return None
+        return indx
+
+    def tip_state(self, tipName):
+        """
+        Return Cartesian endpoint state for a given tip name
+
+        @rtype: EndpointState message
+        @return: pose, velocity, effort
+        """
+        indx = _tip_indx(tipName)
```

Comment:
Seems this could be condensed a bit:
```
try:
    return deepcopy(self._tip_states.states.index(tip_name)) 
except ValueError:
    return None
```